mne/viz/ica.py

```diff
@@ -213,13 +297,12 @@ def plot_ica_properties(ica, inst, picks=None, axes=None, dB=True,
     topomap_args = dict() if topomap_args is None else topomap_args
     image_args = dict() if image_args is None else image_args
     image_args["ts_args"] = dict(truncate_xaxis=False, show_sensors=False)
-    for d in (psd_args, topomap_args, image_args):
-        if not isinstance(d, dict):
-            raise ValueError('topomap_args, image_args and psd_args have to be'
-                             ' dictionaries, got %s instead.' % type(d))
-    if dB is not None and isinstance(dB, bool) is False:
-        raise ValueError('dB should be bool, got %s instead' %
-                         type(dB))
```

Comment:
ValueErrors everywhere.

Reply:
A rough check suggests that we are not consistent:
```
larsoner@bunk:~/custombuilds/mne-python$ git grep ".*TypeError.*must.*" | wc -l
86
larsoner@bunk:~/custombuilds/mne-python$ git grep ".*ValueError.*type.*" | wc -l
107
```
But we should use `TypeError` for these so the associated tests will need to change if using `TypeError` breaks things here.

Reply:
Ok, I'll go over it and set it to TypeError in as many cases as I can stomach.

Eventually.

Reply:
Ok, so this is ~200 fixes. I'd prefer to do a few here, but not all ... Ok?